### PR TITLE
SOC-183: create terraform configuration

### PR DESCRIPTION
**Pull request by jerome-leanspace**
- Added the configuration
- Generified the tenant name
- Plugins have to be downloaded manually for now (see README.md)

Pay attention to:
- Remote Agent (we only have 1 satellite in the configuration communicating with Routes)
- Agents only talks with Ground Station Puertollano: do we want to reduce the number of GS or add them all here? (see the root main.tf)
- Leafspace Ids in `ground_stations/variables.tf`: should we keep them or not?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,10 @@
-# space-operations-center-terraform
\ No newline at end of file
+# space-operations-center-terraform
+
+This repository contains the configuration used for the SOC (Space Operations Center)
+
+Please modify this project with:
+
+- in the root `main.tf` in the provider, input your tenant and a client id and client secret; the last two can be found in a Service Account of the service Teams&Roles
```
Comment (ls-yannick) [inline README.md:7]:
```suggestion
- in the root `main.tf` in the provider, input your tenant, a client id and client secret; the last two can be found in a Service Account of the service [Teams & Roles](https://console.leanspace.io/services/roles/service-accounts)
```

File: ground_stations/main.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
+terraform {
+  required_providers {
+    leanspace = {
+      source = "leanspace/leanspace"
```
Comment (ls-yannick) [inline ground_stations/main.tf:4]: Does the provider get automatically installed?

**Comment by ls-yannick** [comment]
> Agents only talks with Ground Station Puertollano: do we want to reduce the number of GS or add them all here? (see the root main.tf)

I'm fine either way, but I think we could keep only one or two. I'm assume users will want to use their own stations passed the demo stage, so less stations = less things to remove.

> Leafspace Ids in ground_stations/variables.tf: should we keep them or not?

I would say yes. If they are missing, we will need instruction in the docs related to leafspace import to add a line in terraform, and it will be messy. Maybe we can just add a comment, eg. "This tag is only needed if you are importing contacts from leafspace"?

**Review by ls-yannick — APPROVED** [review]
Small suggestions but no blockers :)

File: ground_stations/main.tf (hunk 2 above)
Comment (jerome-leanspace) [reply ground_stations/main.tf:4]: by doing `terraform init` the provider will be automatically installed

**Comment by jerome-leanspace** [comment]
> > Agents only talks with Ground Station Puertollano: do we want to reduce the number of GS or add them all here? (see the root main.tf)
> 
> I'm fine either way, but I think we could keep only one or two. I'm assume users will want to use their own stations passed the demo stage, so less stations = less things to remove.
> 
> > Leafspace Ids in ground_stations/variables.tf: should we keep them or not?
> 
> I would say yes. If they are missing, we will need instruction in the docs related to leafspace import to add a line in terraform, and it will be messy. Maybe we can just add a comment, eg. "This tag is only needed if you are importing contacts from leafspace"?

Yes! I reduced the number of satellites and ground stations to 1 and added the comment you suggested :)

File: ground_stations/main.tf (hunk 2 above)
Comment (ls-yannick) [reply ground_stations/main.tf:4]: Ok nice :)

**Pull request merged**